ansible-scylla-node/tasks/Debian_install.yml

```diff
@@ -1,84 +1,57 @@
+      register: aptversions
+      vars:
+        scylla_version_escaped: "{{ scylla_version_to_install | regex_escape }}"
+
+      #  - debug:
+      #      msg: "{{ aptversions.stdout_line }}"
+
+    - name: "Validate scylla version correctness"
+      ansible.builtin.fail:
+        msg: "Too many/few choices for a requested version '{{ scylla_version_to_install }}': {{ aptversions.stdout_lines }}. Bailing out!"
+      when: aptversions.stdout_lines | length != 1
+
+    - name: If Scylla is installed check if it's the same version as requested
+      fail:
+        msg: "Installed version ({{ installed_scylla_version }}) doesn't match a requested one ({{ aptversions.stdout }})"
+      when: scylla_is_installed and not skip_installed_scylla_version_check and installed_scylla_version != aptversions.stdout
+
+    - name: Do install Scylla
+      block:
+        - name: Nuke a {{ scylla_edition }} pin file if exists
           file:
             state: absent
-            path: /etc/apt/preferences.d/99-scylla
-
-        - name: Install latest OSS Scylla
+            path: "/etc/apt/preferences.d/99-{{ scylla_package_prefix }}"
+
+        - name: Fetch version parts of Scylla package
+          set_fact:
+            scylla_version_split: "{{ aptversions.stdout | regex_findall(regexp, ignorecase=True) }}"
+          vars:
+            # All we know that the version is a string comprised of 3 parts separated by '-'
+            regexp: '^([^\-]+)-([^\-]+)-([^\-]+)$'
+
+        - name: Create package version pin file
+          template:
+            src: templates/apt-pin-file.j2
+            dest: "/etc/apt/preferences.d/99-{{ scylla_package_prefix }}"
+            owner: root
+            group: root
+            mode: '0644'
+
+        - name: "Install {{ aptversions.stdout }}"
           apt:
-            name: scylla
-            state: latest
-        when: scylla_edition == 'oss'
-
-      - block:
-        - name: Nuke an Enterprise pin file if exists
-          file:
-            state: absent
-            path: /etc/apt/preferences.d/99-scylla-enterprise
-
-        - name: Install latest Enterprise Scylla
-          apt:
-            name: scylla-enterprise
-            state: latest
-        when: scylla_edition == 'enterprise'
-    when: scylla_version_to_install == 'latest'
-
-### Handle non-latest version installation
-  - name: Install explicitly specified Scylla version
-    block:
-      - name: Set Scylla package prefix as OSS
-        set_fact:
-          scylla_package_prefix: "scylla"
-        when: scylla_edition == 'oss'
-
-      - name: Set Scylla package prefix as Enterprise
-        set_fact:
-          scylla_package_prefix: "scylla-enterprise"
-        when: scylla_edition == 'enterprise'
-
-      - name: Get versions of {{ scylla_edition }} package
```

Comment:
it's hard to read the diffs here
was this just reorganization of blocks or also some rewrite/fixes in them?

Reply:
A lot of deleted pieces mostly since a lot of it moved to main.yaml.

Let me rebase the PR and split this patch into two.

Reply:
Done.